### ISSUE TITLE
Does OctoDNS handle weighted Route53 records?

### ISSUE THREAD

**Issue by lbrachet** [octodns/octodns-route53#47]
Hello OctoDNS,

I'm testing it and it's working great!

I'm wondering if OctoDNS handles weighted Route53 records?

For example, I created 2 records:

```
Record Name           Type      Routing Policy   Differentiator           Value         Record Id
test1.domain.com      A       Weighted                 100                192.169.0.1       (datacenter1)
test1.domain.com      A       Weighted                 0                 192.169.0.2       (datacenter2)
```

1) I can't find a way to generate the right YAML.
2) I used "octodns-dump" in order to dump the 2 records and shows how the Yaml was generetad but the command nevers dumped the records with "weighted".
If my zone has 5 records, octodns-dump sees only 3 records:
```
INFO  Route53Provider[route53] populate:   found 3 records, exists=True
INFO  YamlProvider[dump] plan:   Creates=3, Updates=0, Deletes=0, Existing Records=0
```

Thanks for your help!

**Comment by ross** [comment]
> * I can't find a way to generate the right YAML.

Weighted records are configured in octoDNS with `dyanmic`, see https://github.com/octodns/octodns/blob/main/docs/dynamic_records.md. In this case that'd look something like:

```yaml
test1:
  dynamic:
    pools:
      main:
        values:
          - value: 192.169.0.1
            weight: 10
          - value: 192.169.0.2
            weight: 15
    rules:
      - pool: main
  type: A
  values:
    - 192.169.0.1
    - 192.169.0.2
```

> 2\. I used "octodns-dump" in order to dump the 2 records and shows how the Yaml was generetad but the command nevers dumped the records with "weighted".

octoDNS tries to unify behaviors across providers and thus doesn't support all possible setups so the manually created weighted set won't be imported if it doesn't match how an octoDNS created record would look.

In the case above the `dynamic` section tells octoDNS what to create for providers that support the advanced dynamic features (like Route53.) The plain `values` sections tells it what to do when pushed to providers that don't support the dynamic features.

**Comment by lbrachet** [comment]
Hello Ross,

Thanks a lot for your answer, it helped me!

The way OctoDNS configures R53 is quite different: there is a Routing Policy = Geolocation with Failover Primary / Secondary.

Usually, we used only records with weight (as I mentionned it in my first post) and that's all. 
I need to check if the result is the same ;)

Last question: R53 handles weight from 0 but I can't set this value, I received:

`- invalid weight "0" in pool "main" value 2`

Thanks.
